zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -1038,7 +1095,100 @@ def test_003_object_storage_and_secondary_block(self):
+        primary_endpoint = self.get_rgw_endpoint(self.primary_rgw_unit)
+        self.assertNotEqual(primary_endpoint, None)
+
+        # Create RGW client and perform IO to be synced to both S3 targets.
+        access_key, secret_key = self.get_client_keys()
+        primary_client = boto3.resource("s3",
+                                        verify=False,
+                                        endpoint_url=primary_endpoint,
+                                        aws_access_key_id=access_key,
+                                        aws_secret_access_key=secret_key)
+        default_container = primary_client.Bucket(default_container_name)
+        default_container.create()
+        default_obj = primary_client.Object(default_container_name, obj_name)
+        default_obj.put(Body=default_obj_data)
+        dev_container = primary_client.Bucket(dev_container_name)
+        dev_container.create()
+        dev_obj = primary_client.Object(dev_container_name, obj_name)
+        dev_obj.put(Body=dev_obj_data)
+
+        # Wait for sync to complete.
+        logging.info('Waiting for Cloud Sync Data and Metadata to Synchronize')
+        self.wait_for_status(self.cloud_sync_rgw_app, is_primary=False)
+
+        # Create clients for the cloud-sync S3 targets.
+        default_s3_client = self.get_minio_boto3_client(
+            self.cloud_sync_default_s3_app
+        )
+        self.assertNotEqual(default_s3_client, None)
+        dev_s3_client = self.get_minio_boto3_client(self.cloud_sync_dev_s3_app)
+        self.assertNotEqual(dev_s3_client, None)
+
+        # Verify that data was properly synced.
+        logging.info('Verifying Synced Data on S3 Targets')
+        test_data = self.fetch_rgw_object(default_s3_client,
+                                          default_container_name,
+                                          obj_name)
+        self.assertEqual(test_data, default_obj_data)
+        test_data = self.fetch_rgw_object(dev_s3_client,
+                                          dev_container_name,
+                                          obj_name)
+        self.assertEqual(test_data, dev_obj_data)
+
+        # Perform cleanup.
+        logging.info('Performing Cleanup')
+        self.purge_bucket(self.primary_rgw_app, default_container_name)
+        self.purge_bucket(self.primary_rgw_app, dev_container_name)
+
+        # Wait for sync to complete.
+        self.wait_for_status(self.cloud_sync_rgw_app, is_primary=False)
+
+        # Validate that synced data was removed from the S3 targets.
+        logging.info('Verifying that data was deleted on the S3 targets')
+        with self.assertRaises(botocore.exceptions.ClientError):
+            default_s3_client.Object(default_container_name, obj_name).get()
+        with self.assertRaises(botocore.exceptions.ClientError):
+            dev_s3_client.Object(dev_container_name, obj_name).get()
+
+    def test_100_migration_and_multisite_failover(self):
```

Comment:
Is there any particular reason we renamed the 004 test ? These numbers define the order in which the tests are executed hence I would prefer that the existing order is not changed unless really needed.

Reply:
We want the `migration_and_multisite_failover` test case to be executed last, that's why I renamed it.

I noticed the following comment:
https://github.com/openstack-charmers/zaza-openstack-tests/blob/b0d0012a9559cf1abc8a470eba9379f3e4f6c3d1/zaza/openstack/charm_tests/ceph/tests.py#L649-L652

Therefore, I assumed that due to the unsupported scenario, we want that test to be executed at the end.

Renaming it the prefix from `test_004` to `test_100` would allow us to add more tests between `test_003` and `test_100`. This is for future-proofing, as another test case is added in the PR https://github.com/openstack-charmers/zaza-openstack-tests/pull/1193.

Reply:
Ack, that does sound OK to me.